docs/contribute/test/testing.md

```diff
@@ -0,0 +1,139 @@
+---
+sidebar_position: 1
+---
+
+# Testing
+
+This module will teach you about writing tests for C++ Code for adding test suites for new plugins or modules for WasmEdge Software. This is an important aspect if you want to submit plugins to the WasmEdge community and we would require you to write test suites to cater to the plugin functionalities.
+
+## Writing Tests for C++ Code Targeting WebAssembly
+
+Testing is a crucial part of software development, ensuring that code behaves as expected. In this guide, we'll explore how to write tests for C++ code specifically designed for WebAssembly (Wasm) environments, using Google Test (gtest) and a specific example related to the WasmEdge runtime.
```

Comment:
write tests for C++ code specifically designed for WebAssembly (Wasm) environments --- Still means to compiling C++ code to the Wasm file.

Reply:
Will update it